qutebrowser/utils/urlutils.py

```diff
@@ -298,14 +298,28 @@ def qurl_from_user_input(urlstr):
     else:
         ipstr = urlstr.strip()
         rest = ''
+
+    # This may error out because we first use this function to validate the
+    # options (_validate_all) before the register the config...
+    try:
+        default_scheme = config.get('network', 'scheme-order')[0]
+    except KeyError:
+        default_scheme = None
```

Comment:
Mainly leaving this here so I don't forget:

This will even mask `KeyError`s in unrelated places - say if an empty list is returned and `[0]` would raise a `KeyError`.

To fix this, `objreg` can be changed to raise a more specialized exception:

```diff
diff --git a/qutebrowser/utils/objreg.py b/qutebrowser/utils/objreg.py
index 8038ad2..10d6873 100644
--- a/qutebrowser/utils/objreg.py
+++ b/qutebrowser/utils/objreg.py
@@ -45,6 +45,16 @@ class RegistryUnavailableError(Exception):
     pass
 
 
+class ItemUnavailableError(KeyError):
+
+    """Exception raised when a given item was not found in the registry.
+
+    Inherits KeyError because there might be existing code catching KeyError.
+    """
+
+    pass
+
+
 class NoWindow(Exception):
 
     """Exception raised by last_window if no window is available."""
@@ -215,11 +225,11 @@ def get(name, default=_UNSET, scope='global', window=None, tab=None):
     reg = _get_registry(scope, window, tab)
     try:
         return reg[name]
-    except KeyError:
+    except KeyError as e:
         if default is not _UNSET:
             return default
         else:
-            raise
+            raise ItemUnavailableError from e
 
 
 def register(name, obj, update=False, scope=None, registry=None, window=None,
diff --git a/qutebrowser/utils/urlutils.py b/qutebrowser/utils/urlutils.py
index 6994369..8ada487 100644
--- a/qutebrowser/utils/urlutils.py
+++ b/qutebrowser/utils/urlutils.py
@@ -29,7 +29,7 @@ from PyQt5.QtCore import QUrl
 from PyQt5.QtNetwork import QHostInfo, QHostAddress
 
 from qutebrowser.config import config, configexc
-from qutebrowser.utils import log, qtutils, message, utils
+from qutebrowser.utils import log, qtutils, message, utils, objreg
 from qutebrowser.commands import cmdexc
 
 
@@ -303,7 +303,7 @@ def qurl_from_user_input(urlstr):
     # options (_validate_all) before the register the config...
     try:
         default_scheme = config.get('network', 'scheme-order')[0]
-    except KeyError:
+    except objreg.ItemUnavailableError:
         default_scheme = None
 
     # Then we try to parse it as an IPv6, and if we fail use
```